horovod/ray/runner.py

```diff
@@ -305,35 +234,71 @@ def __init__(self,
         self.use_gpu = use_gpu
         self.gpus_per_slot = gpus_per_slot or 1
 
+        self.workers = []
+        self.placement_group = None
+
     @property
     def num_workers(self):
         return self.num_hosts * self.num_slots
 
     def _create_workers(self, host_resources):
-        colocator_cls = NodeColocator.options(**host_resources)
-        # Create a number of coordinators.
-        colocators = [
-            colocator_cls.remote(
-                node_rank=node_rank,
-                num_slots=self.num_slots,
-                world_size=self.num_workers,
-                use_gpu=host_resources["num_gpus"] > 0)
-            for node_rank in range(self.num_hosts)
-        ]
-        # We must save a pointer to each colocator to prevent their resource
-        # allocation from being released, along with their children from
-        # going out of scope.
-        self.colocators = colocators
-
-        node_ids = map_blocking(lambda a: a.create_workers.remote(),
-                                colocators)
-        if not len(set(node_ids)) == len(node_ids):
-            raise RuntimeError("Colocator actors must "
-                               f"be placed on unique nodes! Got: {node_ids}")
+        bundles = [host_resources.copy() for _ in range(self.num_hosts)]
+        pg = ray.util.placement_group(bundles, strategy="STRICT_SPREAD")
+        self.placement_group = pg
+        logger.debug("Waiting for placement group to start.")
+        ready, _ = ray.wait([pg.ready()],
+                            timeout=int(os.environ.get(
+                                "HOROVOD_PLACEMENT_GROUP_TIMEOUT_S", 100)))
+        if ready:
+            logger.debug("Placement group has started.")
+        else:
+            raise TimeoutError("Placement group creation timed out. Make sure "
+                               "your cluster either has enough resources or use "
+                               "an autoscaling cluster. Current resources "
+                               "available: {}, resources requested by the "
+                               "placement group: {}".format(
+                ray.available_resources(), pg.bundle_specs))
+
+        # Placement group has started. Now create the workers.
+        self.workers = []
+        # Keep ref of one worker per node for NIC detection.
+        node_workers = []
+        # STRICT_SPREAD guarantees each bundle is on a different node.
+        # Create num_slots workers per bundle, i.e. per machine.
+        for bundle_index in range(len(bundles)):
+            gpu_id_futures = []
+            curr_node_workers = []
+            for i in range(self.num_slots):
+                remote_cls = ray.remote(BaseHorovodWorker)
```

Comment:
can you move this `ray.remote` call outside of the forloop? It only needs to be called once

Reply:
yes good point! Fixed